[s] input checking

actioninja opened the pull request:
Sanity checks + flood checking.

Changes keys_held to a rolling buffer instead of a list of arbitrary size. This limits held keys to 15, I recommend not holding keys with your toes personally.

MrPerson commented:
Try triggering it with an autohotkey script

actioninja commented:
It works.

deathride58 commented:
Checking if the key exists in a list seems like quite a bit of overhead for something that could be solved with either a spam check or a length check, and also wouldn't be too friendly to byond updates if it's made to recognize new keys. This solution is pretty overkill. If you want to test it yourself to make sure your fix works, disable the hotkeys in skin.dmf and add a single +REP hotkey that passes a thousand character long string to keydown

1fbff5f83b23d39d38b1dfcb4cac8d9b commented:
I think this would allow you to send 20 49-characters long strings and spam the hell out of admins without getting kicked

optimumtact commented on code/modules/keybindings/bindings_client.dm:
```diff
@@ -1,9 +1,151 @@
+			QDEL_IN(src, 1)
+			return
+		//No matter what, reject the message.
+		log_admin("Client [ckey] just attempted to send an invalid keypress.  Keymessage was \"[_key]\", report this!.")
```
until we know how likely this is I'd suggest just autokicking to preveent spam

I can't hink of very many cases where it will inconvenience people beyond having to rejoin

actioninja committed:
input checking

commenting + defines

stronger validation + order change so spam checking always happens

more repetitive code to avoid an assignment

rolling buffer keys_held

comments + small code improvement

another comment

autodoc comments

Kmc2000 commented:
based

Kurgis commented:
People are merging this thing before it's even done.
